integration-tests/src/bin/start_mock_network.rs

```diff
@@ -19,15 +19,30 @@ use std::time::Duration;
 /// Program to start a testing environment for one client and a mock network environment
 /// The mock network simulates the entire network by reading a pre-generated chain history
 /// on storage and responds to the client's network requests.
-/// The binary runs in two modes, Sync and NoSync, determined by flag `sync`.
+/// The binary runs in two modes, NoNewBlocks and ProduceNewBlocks, determined by flag `mode`.
 ///
-/// In Sync mode, the client and mock network start from different height and client is trying
-/// to catch up. The client starts from genesis height, and the mock network starts from
-/// target height (see args.target_height) and no new blocks are produced.
+/// In NoNewBlocks mode, the mock network will not simulate block production.
+/// The simulated peers will start from the target height and no new blocks are produced.
 ///
-/// In NoSync mode, the client and mock network start at the same height and new blocks will
-/// be produced. They both start from genesis height and the mock network will produce
-/// blocks until target height.
+/// In ProduceNewBlocks mode, new blocks will be produced, i.e., the client will receive new
+/// blocks from the mock network. User can provide a number in this mode, which will be the starting
+/// height the new produced blocks. If no number is provided, the mock network will start at
+/// the same height as the client, which is specified by client_start_height.
```

Comment:
What are in those blocks? Are they empty or filled with transactions? If they are not empty how are the blocks produced?

Reply:
These are from an existing storage. The mock network takes an existing data dir from a node. It reads the chain history from that source data dir and simulate the generation of new blocks by feeding these blocks to the client.